src/Lachain.Core/Consensus/EraBroadcaster.cs

```diff
@@ -108,6 +115,101 @@ public void Broadcast(ConsensusMessage message)
             }
         }
 
+        public void LoadState(bool restore)
+        {
+            if (!restore)
+            {
+                Logger.LogInformation($"Starting new era {_era}.");
+                _messageEnvelopeRepositoryManager.StartEra(_era, true);
+                return;
+            }
+            
+            _messageEnvelopeRepositoryManager.LoadFromDb();
+            if (!_messageEnvelopeRepositoryManager.IsPresent || _messageEnvelopeRepositoryManager.GetEra() != _era)
+            {
+                Logger.LogWarning($"No outstanding messages from era {_era} found. Starting new era.");
+                _messageEnvelopeRepositoryManager.StartEra(_era);
+            }
+            else
+            {
+                var messages = _messageEnvelopeRepositoryManager.GetMessages().ToList();
+                Logger.LogInformation($"Restoring {messages.Count} Messages from era {_era}");
+                foreach (var messageEnvelope in messages)
```

Comment:
After restoring all messages, manually start thread for all registered protocols.